pyproject.toml

```diff
@@ -19,6 +19,7 @@ requires-python = ">=3.8"
 
 dependencies = [
     "backports.zoneinfo; python_version<'3.9'",
+    "deprecated",
```

Comment:
sigh....

Because this is used in `struct.py`, it is part of the codegen abomination and is thus a build system dep as well

```raw
      File "/Users/runner/work/csp/csp/csp/build/csp_autogen.py", line 17, in <module>
        from csp.impl.struct import Struct  # noqa: E402
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/Users/runner/work/csp/csp/csp/impl/struct.py", line 5, in <module>
        from deprecated import deprecated
    ModuleNotFoundError: No module named 'deprecated'
```

add to the build system deps in the top part of this file.

Reply:
Can you check it looks good now?